### PR TITLE
Use constants for hex character

### PR DESCRIPTION
**Pull request by kashike**
Constant names provided by @EterNityCH and @zachbr.

### CODE REVIEW

**Review by zml2008 — CHANGES_REQUESTED** [review]
constant names should be SCREAMING_SNAKE_CASE

File: api/src/main/java/net/kyori/adventure/text/format/TextColor.java [hunk 1]
```diff
@@ -47,6 +47,19 @@
  * @since 4.0.0
  */
 public interface TextColor extends Comparable<TextColor>, Examinable, RGBLike, StyleBuilderApplicable, TextFormat {
+  /**
+   * The hex character.
+   *
+   * @since 4.14.0
+   */
+  char SCREAMING_SNAKE_CASE1 = '#';
```
Comment (zml2008) [inline api/src/main/java/net/kyori/adventure/text/format/TextColor.java]: maybe `HEX_CHARACTER` and `HEX_CHARACTER_BOXED`? idk

Comment (imDaniX) [reply api/src/main/java/net/kyori/adventure/text/format/TextColor.java]: I think the name should somehow indicate that it's used for colors. `TextColor.HEX_CHARACTER` feels very wrong, like, why it's even there.
As such color specifiers are often referred as "color codes", maybe something like `COLOR_CODE_PREFIX`?

Comment (zml2008) [reply api/src/main/java/net/kyori/adventure/text/format/TextColor.java]: I mean, it is in `TextColor` -- maybe `HEX_PREFIX` though?

**Review by zml2008 — APPROVED** [review]

**Pull request merged**